File: product-portal-backend/src/main/java/com/tmdt/group8/entity/Product.java
package com.tmdt.group8.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
public class Product extends BaseEntity {
    private String name;
    private double price;
    private String slug;

    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "id", nullable = false)
    private Category category;

    @ManyToOne
    @JoinColumn(name = "store_id", referencedColumnName = "id", nullable = false)
    private Store store;

    @OneToMany(mappedBy = "product",cascade = CascadeType.ALL)
    private List<UrlProduct> productUrls;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Image> productImages;

}

File: product-portal-fronted/src/models/Product.ts
import BaseResponse from "./BaseResponse";
import {StoreResponse} from "./Store";

export interface ProductResponse extends BaseResponse {
    name: string
    price: string
    slug: string
    category: string
    storeOwner: StoreResponse
    productUrls: string[]
    productImages: string[]
    categoryId: number
}

export interface ProductRequest {
    name: string
    price: number
    categoryId?: number
    productUrl: string
    productImages: string[]
    userId: number
    oldImage?: string[]
}